include/alpaka/atomic/AtomicAtomicRef.hpp

```diff
@@ -0,0 +1,239 @@
+                {
+                    result = old;
+                    result = std::max(result, value);
+                }
+                return old;
+            }
+        };
+
+        //! The CPU threads accelerator AtomicExch.
+        template<typename T, typename THierarchy>
+        struct AtomicOp<AtomicExch, AtomicAtomicRef, T, THierarchy>
+        {
+            ALPAKA_FN_HOST static auto atomicOp(
+                AtomicAtomicRef const& atomic,
+                T* const addr,
+                T const& value) -> T
+            {
+                static_assert(std::is_trivially_copyable_v<T> && boost::atomic_ref<T>::required_alignment <= alignof(T), "Type not supported by AtomicAtomicRef, please use AtomicStdLibLock for your accelerator.");
+                boost::atomic_ref<T> ref(*addr);
+                T old = ref;
+                T result = value;
+                while(!ref.compare_exchange_weak(old, result))
+                {
+                    result = value;
+                }
+                return old;
+            }
+        };
+
+        //! The CPU threads accelerator AtomicInc.
+        template<typename T, typename THierarchy>
+        struct AtomicOp<AtomicInc, AtomicAtomicRef, T, THierarchy>
+        {
+            ALPAKA_FN_HOST static auto atomicOp(
+                AtomicAtomicRef const& atomic,
+                T* const addr,
+                T const& value) -> T
+            {
+                static_assert(std::is_trivially_copyable_v<T> && boost::atomic_ref<T>::required_alignment <= alignof(T), "Type not supported by AtomicAtomicRef, please use AtomicStdLibLock for your accelerator.");
+                boost::atomic_ref<T> ref(*addr);
+                T old = ref;
+                T result = ((old >= value) ? 0 : static_cast<T>(old + 1));
+                while(!ref.compare_exchange_weak(old, result))
+                {
+                    result = ((old >= value) ? 0 : static_cast<T>(old + 1));
+                }
+                return old;
+            }
+        };
+
+        //! The CPU threads accelerator AtomicDec.
+        template<typename T, typename THierarchy>
+        struct AtomicOp<AtomicDec, AtomicAtomicRef, T, THierarchy>
+        {
+            ALPAKA_FN_HOST static auto atomicOp(
+                AtomicAtomicRef const& atomic,
+                T* const addr,
+                T const& value) -> T
+            {
+                static_assert(std::is_trivially_copyable_v<T> && boost::atomic_ref<T>::required_alignment <= alignof(T), "Type not supported by AtomicAtomicRef, please use AtomicStdLibLock for your accelerator.");
+                boost::atomic_ref<T> ref(*addr);
+                T old = ref;
+                T result = ((old >= value) ? 0 : static_cast<T>(old + 1));
+                while(!ref.compare_exchange_weak(old, result))
+                {
+                    result = ((old >= value) ? 0 : static_cast<T>(old + 1));
+                }
+                return old;
```

Comment:
Copy paste error?
```suggestion
                boost::atomic_ref<T> ref(*addr);
                T old = ref;
                T result = ((old >= value) ? 0 : static_cast<T>(old - 1));
                while(!ref.compare_exchange_weak(old, result))
                {
                    result = ((old >= value) ? 0 : static_cast<T>(old - 1));
                }
                return old;
```
And maybe this suffices:
```suggestion
                return ref.fetch_sub(T{1});
```